### PR TITLE
feat(module:statistic): add statistic and countdown component

### PR DESCRIPTION
**Pull request by wendzhue**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2754 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/statistic/nz-countdown.component.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      this.updater_ = null;
+    }
+  }
+
+  /**
+   * Update time that should be displayed on the screen.
+   * @override
+   */
+  protected updateValue(): void {
+    this.diff = Math.max(this.target - (new Date()).getTime(), 0);
```
[the remark this answers is not in the document]
Comment (wendzhue) [reply components/statistic/nz-countdown.component.ts]: That's funny. 😆

File: components/statistic/nz-countdown.component.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  templateUrl    : './nz-countdown.component.html'
+})
+export class NzCountdownComponent extends NzStatisticComponent implements OnInit, OnChanges, OnDestroy {
+  /** @override */
+  @Input() nzFormat: string = 'HH:mm:ss';
+
+  diff: number;
+  displayText = '';
+  target: number;
+  updater_: Subscription;
```
Comment (HsuanXyz) [inline components/statistic/nz-countdown.component.ts]:
```ts
 updater_ = new Subscription();
```

File: components/statistic/nz-statistic-number.component.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    'class': 'ant-statistic-content-value'
+  },
+  styles             : [ 'nz-number { display: inline }' ]
+})
+export class NzStatisticNumberComponent implements OnChanges {
+  @Input() nzValue: NzStatisticValueType;
+  @Input() nzValueTemplate: TemplateRef<{ $implicit: NzStatisticValueType }>;
+  @Input() nzPrecision: number;
+  @Input() nzDecimalSeparator: string = '.';
+  @Input() nzGroupSeparator: string = ',';
```
Comment (HsuanXyz) [inline components/statistic/nz-statistic-number.component.ts]:
```ts
@Input() nzDecimalSeparator = '.';
@Input() nzGroupSeparator = ',';
```

File: components/statistic/nz-statistic.component.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  @Input() nzPrefix: string | TemplateRef<void>;
+  @Input() nzSuffix: string | TemplateRef<void>;
+  @Input() nzTitle: string | TemplateRef<void>;
+  @Input() nzValue: NzStatisticValueType;
+  @Input() nzValueStyle = {};
+  @Input() nzDecimalSeparator: string = '.';
+  @Input() nzGroupSeparator: string = ',';
+  @Input() nzValueTemplate: TemplateRef<{ $implicit: NzStatisticValueType }>;
+
+  constructor(protected cdr: ChangeDetectorRef) {
```
Comment (HsuanXyz) [inline components/statistic/nz-statistic.component.ts]: 这个是不是不用注入？

**Comment by wendzhue** [comment]
Since we are going to change date format string API, this PR should be ready to merge.

**Pull request merged**